modules/terraform/workspace.go

```diff
@@ -60,3 +60,49 @@ func nameMatchesWorkspace(name string, workspace string) bool {
 	match, _ := regexp.MatchString(fmt.Sprintf("^\\*?\\s*%s$", name), workspace)
 	return match
 }
+
+// WorkspaceDelete removes the specified terraform workspace with the given options.
+// It returns the name of the current workspace AFTER deletion, and the returned error (that can be nil).
+// If the workspace to delete is the current one, then it tries to switch to the "default" workspace.
+// Deleting the workspace "default" is not supported.
+func WorkspaceDeleteE(t testing.TestingT, options *Options, name string) (string, error) {
+	currentWorkspace := RunTerraformCommand(t, options, "workspace", "show")
+
+	if name == "default" {
+		return currentWorkspace, fmt.Errorf("Deleting the workspace 'default' is not supported")
```

Comment:
Please define a custom error type for this instead of using `fmt.Errorf`. That makes it possible to check returned errors by type, rather than string messages that may change.

Reply:
This applies here and to all errors below.

Reply:
There are multiple ways to do it. The answers in [this SO thread](https://stackoverflow.com/questions/23796543/go-checking-for-the-type-of-a-custom-error) provide a good list of the options.

Reply:
Missed this one. Updating to test on the error type with `assert.True(t, errors.As(gotErr, &testCase.expectedError))`